project/src/main/java/com/google/autograder/servlets/CreateNewAssignmentServlet.java

```diff
@@ -34,78 +34,64 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
-
-        values[0] = Integer.parseInt(date.substring(firstSeperator + 1, secondSeperator));
-        values[1] = Integer.parseInt(date.substring(secondSeperator + 1));
-        values[2] = Integer.parseInt(date.substring(0, firstSeperator));
-
-        return values;
-    }
+            String json = API.getJSON(connection);
 
-    private int[] getHoursMinutes(String time) {
-        int seperator = time.indexOf(':');
-        int[] values = new int[2];
+            System.out.println("\n\n" + json + "\n\n");
 
-        values[0] = Integer.parseInt(time.substring(0, seperator));
-        values[1] = Integer.parseInt(time.substring(seperator + 1));
+            responseCode = connection.getResponseCode();
+        } catch(IOException e) {
+            // TODO: Handle error response
+            responseCode = connection.getResponseCode();   
+        }
 
-        return values;
+        response.setHeader("responseCode", String.valueOf(responseCode));
     }
 
     private String buildPostBody(HttpServletRequest request) throws UnsupportedEncodingException {
+        String title = request.getParameter("title");
         String description = request.getParameter("description");
         String maxPoints = request.getParameter("maxPoints");
         String dueDate = request.getParameter("dueDate");
-        String dueTime = request.getParameter("dueTime");
-        String title = request.getParameter("title");
-
-        int[] dayMonthYear = getDayMonthYear(dueDate);
-        int[] hoursMinutes = getHoursMinutes(dueTime);
-
-        StringBuilder postBody = new StringBuilder();
-
-        postBody.append("{");
-        postBody.append("\"title\":\"" + title + "\",");
-        postBody.append("\"description\":\"" + description + "\",");
-        postBody.append("\"maxPoints\":" + maxPoints + ",");
-        postBody.append("\"state\":\"" + "DRAFT" + "\",");
-        postBody.append("\"workType\":\"" + "ASSIGNMENT" + "\",");
-        postBody.append("\"assigneeMode\":\"" + "ALL_STUDENTS" + "\",");
-        postBody.append("\"submissionModificationMode\":\"" + "MODIFIABLE_UNTIL_TURNED_IN" + "\",");
-        postBody.append("\"dueDate\":{");
-        postBody.append("\"day\":" + dayMonthYear[0] + ",");
-        postBody.append("\"month\":" + dayMonthYear[1] + ",");
-        postBody.append("\"year\":" + dayMonthYear[2]);
-        postBody.append("},");
-        postBody.append("\"dueTime\":{");
-        postBody.append("\"hours\":" + hoursMinutes[0] + ",");
-        postBody.append("\"minutes\":" + hoursMinutes[1]);
-        postBody.append("}");
-        postBody.append("}");
-
-        return postBody.toString();
+        String year = request.getParameter("year");
+        String month = request.getParameter("month");
+        String day = request.getParameter("day");
+        String hours = request.getParameter("hours");
+        String minutes = request.getParameter("minutes");
+
+        JsonObject assignmentObject = new JsonObject();
```

Comment:
we have an assignment class currently, can you update that class and make the assignment object with that class? for consistency

Reply:
Right here I'm building out the post body for the create new assignment endpoint. All this data is sent to the Google Classroom API and not to the Database. When this step is completed we redirect to the assignments page where another servlet pulls out the data we need on our end and then adds it to the Database.

Reply:
ahhh got it, makes sense